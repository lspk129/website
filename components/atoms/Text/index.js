import { createElement } from 'react';
import styled from 'styled-components';

const Text = styled(({ tag, hero, upperCase, children, ...props }) => createElement(tag, props, children))`
  text-align: center;

  ${({ tag, theme }) => tag === 'a' && `
    text-decoration: none;
    cursor: pointer;
    color: inherit;
    &:hover { color: ${theme.main}}
  `}

  ${({ tag, hero, theme }) => tag === 'h1' && `
    ${hero && `
      color: ${theme.text};
      font-size: 4rem;
      white-space: nowrap;
      font-weight: bold;
      text-shadow: 0 0.1rem 0.1rem rgba(0, 0, 0, 0.4);
      line-height: 4rem;
    `}
  `}

  ${({ tag, hero, theme }) => tag === 'h2' && `
    ${hero && `
      color: ${theme.text};
      text-shadow: 0 0.1rem 0.1rem rgba(0, 0, 0, 0.2);
    `}
  `}

  ${({ upperCase }) => upperCase && 'text-transform: uppercase'};
`;

Text.defaultProps = {
  tag: 'span',
};

export default Text;
