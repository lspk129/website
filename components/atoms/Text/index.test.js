import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Text from './';

describe('Text', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Text />);
    expect(wrapper).toBePresent();
    expect(wrapper).toHaveProp('tag', 'span');
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders with tag prop h1', () => {
    const wrapper = shallow(<Text tag={'h1'} />);
    expect(wrapper).toHaveProp('tag', 'h1');
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders with hero boolean props', () => {
    const wrapper = shallow(<Text hero />);
    expect(wrapper).toHaveProp('hero', true);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders with upperCase props', () => {
    const wrapper = shallow(<Text upperCase />);
    expect(wrapper).toHaveProp('upperCase', true);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
