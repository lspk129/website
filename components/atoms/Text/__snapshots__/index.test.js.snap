// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders correctly 1`] = `
<Component
  className="sc-bdVaJa jUDgql"
  tag="span"
/>
`;

exports[`Text renders with hero boolean props 1`] = `
<Component
  className="sc-bdVaJa jUDgql"
  hero={true}
  tag="span"
/>
`;

exports[`Text renders with tag prop h1 1`] = `
<Component
  className="sc-bdVaJa jUgipb"
  tag="h1"
/>
`;

exports[`Text renders with upperCase props 1`] = `
<Component
  className="sc-bdVaJa eBqvIR"
  tag="span"
  upperCase={true}
/>
`;
